src/main/java/team4/footwithme/resevation/service/ReservationServiceImpl.java

```diff
@@ -1,9 +1,108 @@
 package team4.footwithme.resevation.service;
 
 import lombok.RequiredArgsConstructor;
+import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import team4.footwithme.chat.service.event.ReservationMemberJoinEvent;
+import team4.footwithme.chat.service.event.ReservationMembersJoinEvent;
+import team4.footwithme.chat.service.event.ReservationPublishedEvent;
+import team4.footwithme.member.domain.Gender;
+import team4.footwithme.member.domain.Member;
+import team4.footwithme.member.repository.MemberRepository;
+import team4.footwithme.resevation.domain.Mercenary;
+import team4.footwithme.resevation.domain.Participant;
+import team4.footwithme.resevation.domain.ParticipantRole;
+import team4.footwithme.resevation.domain.Reservation;
+import team4.footwithme.resevation.repository.MercenaryRepository;
+import team4.footwithme.resevation.repository.ParticipantRepository;
+import team4.footwithme.resevation.repository.ReservationRepository;
+import team4.footwithme.stadium.domain.Court;
+import team4.footwithme.stadium.repository.CourtRepository;
+import team4.footwithme.team.domain.Team;
+import team4.footwithme.team.repository.TeamRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 @RequiredArgsConstructor
-public class ReservationServiceImpl implements ReservationService{
+public class ReservationServiceImpl implements ReservationService {
+
+    private final ReservationRepository reservationRepository;
+    private final CourtRepository courtRepository;
+    private final MemberRepository memberRepository;
+    private final TeamRepository teamRepository;
+    private final ParticipantRepository participantRepository;
+    private final MercenaryRepository mercenaryRepository;
+    private final ApplicationEventPublisher eventPublisher;
+
+    @Transactional
+    @Override
+    public void createReservation(Long memberId, Long courtId, Long teamId, LocalDateTime matchDate, List<Long> memberIds) {
+        Court court = courtRepository.findActiveById(courtId).orElseThrow(
+            () -> new IllegalArgumentException("해당하는 구장이 없습니다.")
+        );
+        Member member = memberRepository.findActiveById(memberId).orElseThrow(
+            () -> new IllegalArgumentException("해당하는 회원이 없습니다.")
+        );
+        Team team = teamRepository.findById(teamId).orElseThrow(
+            () -> new IllegalArgumentException("해당하는 팀이 없습니다.")
+        );
+
+        List<Member> participantMembers = memberRepository.findAllById(memberIds);
+
+        boolean allMale = participantMembers.stream()
```

Comment:
코딩 컨벤션을 지킨다면 다음과 같이 바뀝니다.
```suggestion
        boolean isAllMale = participantMembers.stream()
```
저도 빨리 개발해야하는 마음에 이런 실수 많이 했을 것 같네요...

Reply:
그렇네요 ㅋㅋㅎㅎ 그냥 급하게 하다보니 인텔리제이가 해주는대로 냅뒀더니 ㅠㅠ
지금이라도 수정하겠습니다 감사합니다